slice-test/slice-core-test/src/test/groovy/com/cognifide/slice/api/tag/SliceTagUtilsTest.groovy

```diff
@@ -0,0 +1,152 @@
+
+/**
+ * @author Jaromir Celejewski
+ * Date: 10.04.15
+ */
+class SliceTagUtilsTest extends ProsperSpec{
+
+    @Shared
+    InjectorHierarchy injectorHierarchy
+
+    @Shared
+    InjectorsRepositoryService repositoryService
+
+    @Shared
+    protected Injector injector1
+
+    @Shared
+    protected Injector injector2
+
+    @Shared
+    protected ModelProvider modelProvider
+
+    private static final String BUNDLE_NAME_FILTER = "slice-test-app.*";
+
+    private static final String BASE_PACKAGE = "com.cognifide.example";
+
+    private static final String INJECTOR_NAME = "slice-test";
+
+    def setup() {
+        ContextScope contextScope = new SliceContextScope()
+        List<Module> modules = new ArrayList<Module>()
+        modules.add(new SliceModule(contextScope, null))
+        modules.add(new SlingModule(contextScope))
+        modules.add(new JcrModule())
+        modules.add(new MapperModule())
+        modules.add(new SliceResourceModule())
+        modules.add(new TestModule())
+
+        final InjectorRunner injectorRunner1 = new InjectorRunner(null,
+                "slice-test",
+                "slice-test-app.*",
+                "com.cognifide.example")
+
+        injectorRunner1.installModules(modules)
+        InjectorConfig config = new InjectorConfig(injectorRunner1)
+
+        final InjectorRunner injectorRunner2 = new InjectorRunner(null,
+                "slice-test2",
+                "slice-test-app2.*",
+                "com.cognifide.example2")
+
+        injectorRunner2.installModules(modules)
+        injectorRunner2.setParentInjectorName("slice-test")
+        InjectorConfig config2 = new InjectorConfig(injectorRunner2)
+
+        final InjectorRunner injectorRunner3 = new InjectorRunner(null,
+                "slice-test/subtest",
+                "slice-test-app3.*",
+                "com.cognifide.example3")
+
+        injectorRunner3.installModules(modules)
+        injectorRunner3.setParentInjectorName("slice-test")
+        InjectorConfig config3 = new InjectorConfig(injectorRunner3)
+
+        repositoryService = new InjectorsRepositoryService()
+        injectorHierarchy = new InjectorHierarchy()
+        injectHierarchyIntoRepositoryService(repositoryService, injectorHierarchy)
+
+        Method bindConfigMethod = injectorHierarchy.getClass().getDeclaredMethod("bindConfig", InjectorConfig.class)
+        bindConfigMethod.setAccessible(true)
+        bindConfigMethod.invoke(injectorHierarchy, config)
+        bindConfigMethod.invoke(injectorHierarchy, config2)
+        bindConfigMethod.invoke(injectorHierarchy, config3)
+
+        //ContextFactory factory = injector2.getInstance(ContextFactory.class)
+        //Context context = factory.getResourceResolverContext(resourceResolver)
+        //contextScope.setContextProvider(new ConstantContextProvider(context))
+
+        //modelProvider = injector.getInstance(ModelProvider.class)
+    }
+
+    def "Has injector"() {
```

Comment:
Refactor this. It's nor readable and also seems testing too much as for one test